cmd/osctl/cmd/kubeconfig.go

```diff
@@ -56,11 +60,19 @@ Kubeconfig will be written to PWD/kubeconfig or [local-path]/kubeconfig if speci
 				localPath = args[0]
 			}
 
+			//Drop the existing kubeconfig before writing the new one if force flag is specified.
+			if force {
+				err = os.Remove(filepath.Clean(localPath + "/kubeconfig"))
+				if err != nil && !os.IsNotExist(err) {
+					helpers.Fatalf("error deleting existing kubeconfig: %s", err)
+				}
+			}
 			extractTarGz(filepath.Clean(localPath), r)
```

Comment:
I feel this whole thing better to be rewritten without `extractTarGz`, as this a bit of overkill for a single file

Reply:
we have https://github.com/talos-systems/talos/blob/master/cmd/osctl/pkg/client/client.go#L166 wrapper method which can be used to simply write down `kubeconfig` where appropriate (and hopefully one day merge it into user's kubeconfig)

Reply:
Seems fair, but sounds like it's a bit larger overhaul of this kubeconfig command. I'll throw it in the backlog.

Reply:
Wonder if the new read api would be better here? It supports streaming. But agree something for a follow-up.